test/fake_http_server.cc

```diff
@@ -27,20 +33,101 @@ void FakeServer::SetResponse(const std::string& path,
   handler_.path_responses[path] = response;
 }
 
+void FakeServer::AllowStream(const std::string& path) {
+  handler_.path_streams[path] = std::unique_ptr<Handler::Stream>(
+      new Handler::Stream);
+}
+
+void FakeServer::WaitForOneStreamWatcher(const std::string& path) {
+  auto s = handler_.path_streams.find(path);
+  if (s == handler_.path_streams.end()) {
+    LOG(ERROR) << "No stream for path " << path;
+    return;
+  }
+  auto& stream = s->second;
+  std::unique_lock<std::mutex> lk(stream->mutex);
```

Comment:
Same general comment about `lk`, I'm not sure what we call it elsewhere, as long as we're consistent with how we name locks, that'd be ideal.